client/src/components/addSourceWizard/addSourceWizardStepTwo.js

```diff
@@ -125,19 +122,22 @@ class AddSourceWizardStepTwo extends React.Component {
       credentials.length &&
       !credentialsError
     ) {
-      let updatedSource = {
-        [apiTypes.API_SOURCE_NAME]: sourceName,
-        [apiTypes.API_SOURCE_HOSTS]: hosts,
-        [apiTypes.API_SOURCE_CREDENTIALS]: credentials
-      };
-
+      let updatedSource = { id: source.id };
+      _.set(updatedSource, apiTypes.API_SOURCE_TYPE, sourceType);
+      _.set(updatedSource, apiTypes.API_SOURCE_NAME, sourceName);
+      _.set(updatedSource, apiTypes.API_SOURCE_HOSTS, hosts);
+      _.set(updatedSource, apiTypes.API_SOURCE_SAT_VERSION, satelliteVersion);
+      _.set(updatedSource, apiTypes.API_SOURCE_SSL_PROTOCOL, sslProtocol);
+      _.set(updatedSource, apiTypes.API_SOURCE_SSL_CERT, sslCertVerify);
+      _.set(updatedSource, apiTypes.API_SOURCE_SSL_DISABLE, disableSsl);
+      _.set(updatedSource, apiTypes.API_SOURCE_CREDENTIALS, credentials);
```

Comment:
This partially occurred to me before, viewing this update I've had a reoccurrence... I think we can definitely look at focusing the steps so they only update the properties they're responsible for, similar to using the former `_.merge` that was on `line 140`. If we handle it that way we might help alleviate some muddled confusion around 

> Why are we aiming to re-map the source object, both in resetting the state and in passing it back to Redux... what if we narrowed the scope even more, or at least until step 2 ends up handling the remaining properties. That ties into only having to observe each prior steps updated properties, similar to the "step valid" property.

I think if we combined this concept, with some form of a validation component that simply returns a value towards the state source object/model, and internally handles the validation, we could cut the wizard steps down even further.

Reply:
Yeah, basically what this is doing. I can remove the fields that are not updated by this page and merge instead.